pkg/bundle/publish/manifests.go

```diff
@@ -57,14 +67,11 @@ func GenerateICSP(out io.Writer, name string, byteLimit int, icspScope string, m
 				"operators.openshift.org/catalog": "true",
 			},
 		},
-		Spec: operatorv1alpha1.ImageContentSourcePolicySpec{
-			RepositoryDigestMirrors: []operatorv1alpha1.RepositoryDigestMirrors{},
-		},
 	}
 
 	for key := range registryMapping {
 		icsp.Spec.RepositoryDigestMirrors = append(icsp.Spec.RepositoryDigestMirrors, operatorv1alpha1.RepositoryDigestMirrors{
-			Source:  name,
+			Source:  key,
```

Comment:
@estroz Question here. I changed this from `key` to `name` so the `source` on the ICSP would reflect original image name. The key in the mapping is the location on disk so the source is coming out like this 
```  repositoryDigestMirrors:
  - mirrors:
    - localhost:5001/baz-operator/baz
    source: baz-operator/baz
```
Is this correct for catalogs?

Reply:
It's almost correct; using `name` won't work sometimes because that name may contain a tag, and ICSP sources should not contain a tag or digest afaik. ICSP's should look like:

```yaml
  - mirrors:
    - localhost:5001/baz-operator/baz
    source: quay.io/baz-operator/baz
```

I think we need to add one more component to the association type for the pre-mirror fq image name, which I'll do in a follow-up. Thoughts?

Reply:
We are actually capturing that with the key in the AssociationSet. Checkout this line, there may be better ways to do this, but I am only adding an association to the `icspMapping` is the tag field is empty. https://github.com/RedHatGov/bundle/blob/74677b6fa3af6d899eec816ff95306494047cfc6/pkg/bundle/publish/publish.go#L318

Reply:
Actually we could just use `name` but would need to parse it as to remove tag/digest. I'll whip something up.